pcdsdevices/epics/valve.py

```diff
@@ -279,6 +269,9 @@ def open(self, wait=False, timeout=None, **kwargs):
         return super().open(wait=wait, timeout=timeout, **kwargs)
 
 
+MPSGateValve = partial(mps_factory, 'MPSGateValve', GateValve)
+MPSStopper   = partial(mps_factory, 'MPSStopper', Stopper)
```

Comment:
I'm not 100% sure I like how this comes together down here, but I don't think I have any better alternatives. The other way would have been to have a subclassed `__init__` method that sets the `_mps_prefix` attribute and then have a fixed `{self._mps_prefix}` prefix in the FormattedComponent. This would let mps_factory be used directly to make a class but it makes the implementation of mpg_factory more messy.

Reply:
Yeah I wanted to avoid subclassing `__init__`. Seemed like it could get messy. If we find we need explicit class creation I can go back and change it, but for now I'd argue for the cleaner implementation